.github/workflows/get_rocks_modified_and_build_scan_test.yaml

```diff
@@ -0,0 +1,51 @@
+name: Get ROCKs modified and build-scan-test them
+
+on:
+  workflow_call:
+
+jobs:
+  get-all-rock-paths:
+    name: Get all rockcraft.yaml paths
+    runs-on: ubuntu-22.04
+    outputs:
+      paths: ${{ steps.find.outputs.paths}}
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v3
+      - name: Get all rockcraft.yaml paths using find
+        id: find
+        run: |
+          paths=$(find ./ -name "rockcraft.yaml" -printf "%h:\n  - '%h/**'\n" | sed 's/\.\///g')
+          echo "Paths found:"
+          echo "${paths}"
+          # Passing multi-line string to output requires the use of a delimiter
+          echo "paths<<EOF" >> $GITHUB_OUTPUT
+          echo "${paths}" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+  changes:
+    name: Get paths for ROCKs modified by PR
+    needs: get-all-rock-paths
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: read
+    # Set job outputs to values from filter step
+    outputs:
+      rock_paths: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: ${{ needs.get-all-rock-paths.outputs.paths}}
```

Comment:
Not sure what the output of this paths-filter action is, but if it doesn't print to screen adding another step here that just echo's the content of `steps.filter.outputs.changes` might be helpful to anyone using this in future.